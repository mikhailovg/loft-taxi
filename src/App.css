.Header {
  z-index: 1000;
  display: flex;
  background-color: #fff;
  min-height: 70px;
  padding-left: 24px;
  padding-right: 24px;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2), 0px 4px 5px 0px rgba(0,0,0,0.14), 0px 1px 10px 0px rgba(0,0,0,0.12);
}
.HeaderLogo {
  flex-grow: 1;
  margin: 0;
}

.PageTitle {
  text-align: center;
  margin-top: 40px;
}

.Logo {
  height: 40vmin;
}

.Login {
  background-size: cover;
  background-image: url('./login-background.jpg');
  min-height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
}
.LoginColumn {
  max-width: 25%;
  flex-basis: 25%;
}
.LoginForm {
  padding: 44px 60px;
  min-width: 500px;
  margin-top: 48px;
  margin-bottom: 48px;
  border-radius: 4px;
  background-color: #fff;
}
.LoginFormContainer {
  display: flex;
  flex-direction: column;
}
.LoginFormItem {
  text-align: left;
}
.LoginTitle {
  margin-bottom: 30px;
}
.LoginLink {
  display: flex;
  flex-direction: row;
}
.LoginButton {
  width: 100px;
  display: flex;
  align-items: flex-end;
  align-self: flex-end;
  background-color: yellow;
}

.Link {
  margin-left: 4px;
  color: #1473e6;
  cursor: pointer;
  text-decoration: none;
}

.MapContainer {
  z-index: -1;
  position: relative;
}
.Map {
  height: 700px;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  width: 100%;
}

.HeaderLink {
  color: #1473e6;
  cursor: pointer;
  text-decoration: none;
  align-self: center;
  padding-top: 14px;
  padding-bottom: 14px;
}
.Profile {
  background-size: cover;
  background-image: url('./login-background.jpg');
  min-height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
}
.ProfileTitle {
  font-size: 2.125rem;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  font-weight: 400;
  line-height: 1.17;
  letter-spacing: 0.00735em;
  margin: 0px;
}
.FlexCenter {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  align-self: flex-start;
}
.ProfileSubTitle {
  color: rgba(0, 0, 0, 0.54);
  margin-top: 0px;
  margin-bottom: 40px;
  font-size: 1rem;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  font-weight: 400;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

.ProfileFormContainer {
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  box-sizing: border-box;
  align-content: center;
}
.ProfileForm {
  flex-grow: 0;
  max-width: 100%;
  flex-basis: 100%;
}
.ProfileFormCards {
  flex-grow: 1;
  width: calc(100% + 32px);
  margin: -16px;
  justify-content: center;
  display: flex;
  /*flex-wrap: wrap;*/
  box-sizing: border-box;
}

.ProfileFormCardContainer {
  flex-grow: 0;
  max-width: 50%;
  flex-basis: 50%;
  padding: 16px
}

.ProfileFormCard {
  width: 236px;
  height: 157px;
  position: relative;
  padding: 16px 32px;
  border-radius: 4px;
  box-shadow: 0px 3px 3px -2px rgba(0,0,0,0.2), 0px 3px 4px 0px rgba(0,0,0,0.14), 0px 1px 8px 0px rgba(0,0,0,0.12);
}

.ProfileFormCardFields {
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
}
.IconMasterCard {
  top: 8px;
  right: 8px;
  width: 32px;
  position: absolute;
  font-size: 1.5rem;
}
.ProfileFormButton {
  display: flex;
  justify-content: center;
  margin-top: 40px;
}
.TextFieldLabelBlock {
  margin-top: 16px;
  margin-bottom: -16px;
}
.TextFieldLabel {
  color: rgba(0, 0, 0, 0.54);
  padding: 0;
  font-size: 12px;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  font-weight: 400;
  line-height: 1;
  letter-spacing: 0.00938em;
}
.MuiButton-containedPrimary {
  color: rgba(0, 0, 0, 0.87)!important;
  background-color: #ffc617 !important;
}
.MuiButton-contained {
  box-shadow: none !important;
}
.LoginButtonMargin {
  margin-top: 20px !important;
}

.MapFormContainer {
  padding: 44px 60px;
  margin-top: 48px;
  margin-bottom: 48px;
  border-radius: 4px;
  background-color: #fff;

  position: absolute;
  top: 0;
  left: 20px;
  max-width: 30%;

  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2), 0px 1px 1px 0px rgba(0,0,0,0.14), 0px 1px 3px 0px rgba(0,0,0,0.12);
}
.MapForm {
  width: 298px;
  display: flex;
  flex-direction: column;
}
.MapFormRow {
  width: 100%;
  margin-bottom: 40px;
}
.MapFormButton {
  width: 100%;
  margin-top: 0 !important;
  text-transform: none !important;
}
.MuiInput-root {
  width: 100%;
}
.Hidden {
  display: none !important;
}
.MapOrderTitle {
  margin-top: 0;
  margin-bottom: 30px;
  font-size: 2.125rem;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  font-weight: 400;
  line-height: 1.17;
  letter-spacing: 0.00735em;
}
.MapOrderText {
  margin-top: 0;
  margin-bottom: 30px;
  font-size: 1rem;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  font-weight: 400;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}
.ButtonDisabled {
  color: rgba(0, 0, 0, 0.26) !important;
  box-shadow: none !important;;
  background-color: rgba(0, 0, 0, 0.12) !important;;
}
.ProfileSaveSuccess {
  margin-bottom: 40px;
}
.ButtonLowerCase {
  text-transform: none !important;
}
.Error {
  color: #f44336;
  margin: 0;
  font-size: 0.75rem;
  margin-top: 8px;
  min-height: 1em;
  text-align: left;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  font-weight: 400;
  line-height: 1em;
  letter-spacing: 0.03333em;
}