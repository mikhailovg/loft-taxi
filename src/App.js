import React from 'react';
import './App.css';

import {Profile} from './profile';
import {Map} from './map';
import {Login} from './login';
import {Registration} from './registration';
import {Header} from './shared/Header';

import {
    Redirect,
    Route,
    Switch,
} from 'react-router-dom'
import {shallowEqual, useSelector} from 'react-redux'

import {

    getAuth,
    getRegister,

} from './modules/main'
import {Router} from 'react-router'
import history from './history'

function App() {

    const auth = useSelector(getAuth, shallowEqual)
    const register = useSelector(getRegister, shallowEqual)

    return (
        <Router history={history}>
            {(auth && auth.success && JSON.parse(auth.success) === true) || (register && register.success && JSON.parse(register.success) === true) ?
                <Header id={'header'}/> : null
            }
            <Switch>
                <Route path={'/login'} render={() => <Login/>}/>
                <Route path={'/map'} render={() => <Map/>}/>
                <Route path={'/profile'} render={() => <Profile/>}/>
                <Route path={'/registration'} render={() => <Registration/>}/>
                <Redirect from={'/'} to={'login'}/>
            </Switch>

        </Router>
    )
}

export default App
