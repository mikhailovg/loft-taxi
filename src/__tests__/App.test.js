import React from 'react'
import ReactDOM from 'react-dom'
import {mount, shallow} from 'enzyme'
import '../setupTests'
import renderer from 'react-test-renderer'

import App from '../App'

describe('App', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders Header', () => {
        const wrapper = shallow(<App/>);
        expect(wrapper.find('#header').length === 0).toEqual(true);
    });

    it('renders correctly', () => {
        const tree = renderer.create(<App/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})

// FIX TypeError: window.URL.createObjectURL is not a function
jest.mock('mapbox-gl', () => ({
    Map: () => ({})
}))

jest.mock('react-redux', () => ({
    useDispatch: () => {},
    useSelector: () => ({
        auth: {
            success: true,
            token: 'TOKEN1'
        },
        register: {
            success: true,
            token: 'TOKEN1'
        },
    })
}))