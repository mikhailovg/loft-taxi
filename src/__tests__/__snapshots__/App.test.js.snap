// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="Login"
>
  <div
    className="LoginColumn"
  >
    <img
      alt="logo"
      className="Logo"
      src="logo.svg"
      width={156}
    />
  </div>
  <div
    className="LoginColumn"
  >
    <div
      className="LoginForm"
      data-testid="LoginForm"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="LoginFormContainer"
        >
          <div
            className="LoginFormItem"
          >
            <h1
              className="LoginTitle"
            >
              Войти
            </h1>
            <div
              className="LoginLink"
            >
              <div>
                Уже зарегистрированы?
              </div>
              <a
                className="Link"
                href="/registration"
                onClick={[Function]}
              >
                Зарегистрируйтесь
              </a>
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
              data-shrink={false}
            >
              Имя пользователя
              <span
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                name="login"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Имя пользователя"
                required={true}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
              data-shrink={false}
            >
              Пароль
              <span
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                name="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Пароль"
                required={true}
                type="password"
                value=""
              />
            </div>
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained LoginButton LoginButtonMargin MuiButton-containedPrimary"
            data-testid="LoginButton"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Войти
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
